### PR TITLE
Add trace token support to HTTP client

### PR DESCRIPTION
**Pull request by nezihyigitbasi**
Built on top of https://github.com/airlift/airlift/pull/606

This will be useful to correlate requests in the client/server logs as discussed with @dain.

### CODE REVIEW

File: http-client/src/test/java/io/airlift/http/client/jetty/TestClientTracing.java [hunk 1]
```diff
@@ -0,0 +1,124 @@
...
+    }
+
+    @AfterMethod(alwaysRun = true)
+    public void tearDown()
+            throws Exception
+    {
+        if (server != null) {
+            server.stop();
+        }
+        Files.walk(tmpLogPath)
```
Comment (dain) [inline http-client/src/test/java/io/airlift/http/client/jetty/TestClientTracing.java]: I think we have a util for this somewhere (@electrum?)

File: http-client/src/test/java/io/airlift/http/client/jetty/TestClientTracing.java [hunk 2]
```diff
@@ -0,0 +1,124 @@
...
+                            config.setLogPath(tmpLogPath.toAbsolutePath().toString());
+                        }),
+                new TraceTokenModule())
+                .quiet()
+                .strictConfig()
+                .initialize();
+        JettyHttpClient httpClient = (JettyHttpClient) injector.getInstance(Key.get(HttpClient.class, TestClient.class));
+        int port = ((ServerConnector) server.getConnectors()[0]).getLocalPort();
+        URI uri = new URI("http", null, "127.0.0.1", port, null, null, null);
+        StringResponse response = httpClient.execute(Request.Builder.prepareGet().setUri(uri.resolve("/test")).build(), createStringResponseHandler());
```
Comment (dain) [inline http-client/src/test/java/io/airlift/http/client/jetty/TestClientTracing.java]: You could change the test servlet to return the token text and then verify it below.

File: http-client/src/test/java/io/airlift/http/client/jetty/TestClientTracing.java (hunk 1 above)
Comment (electrum) [reply http-client/src/test/java/io/airlift/http/client/jetty/TestClientTracing.java]: Guava's `MoreFiles`
```java
deleteRecursively(tmpLogPath.toPath(), ALLOW_INSECURE);

File: http-client/src/main/java/io/airlift/http/client/HttpClientModule.java [hunk 3]
```diff
@@ -113,12 +116,19 @@ public HttpClient get()
             KerberosConfig kerberosConfig = injector.getInstance(KerberosConfig.class);
             HttpClientConfig config = injector.getInstance(Key.get(HttpClientConfig.class, annotation));
 
+            boolean clientTracingEnabled = injector.getExistingBinding(Key.get(TraceTokenRequestFilter.class)) != null;
+            Optional<TraceTokenManager> traceTokenManager = Optional.empty();
+            if (clientTracingEnabled) {
+                verify(injector.getExistingBinding(Key.get(TraceTokenManager.class)) != null, "TraceTokenManager is not bound");
```
Comment (electrum) [inline http-client/src/main/java/io/airlift/http/client/HttpClientModule.java]: This doesn't make sense. We asked for this two lines above.

Comment (electrum) [reply http-client/src/main/java/io/airlift/http/client/HttpClientModule.java]: I think we can use `OptionalBinder` for this.

File: http-client/src/main/java/io/airlift/http/client/jetty/JettyHttpClient.java [hunk 4]
```diff
@@ -352,6 +356,7 @@ private static Scheduler createScheduler(String name, int timeoutConcurrency, in
     public <T, E extends Exception> T execute(Request request, ResponseHandler<T, E> responseHandler)
             throws E
     {
+        traceTokenManager.ifPresent(TraceTokenManager::createAndRegisterNewRequestToken);
```
Comment (electrum) [inline http-client/src/main/java/io/airlift/http/client/jetty/JettyHttpClient.java]: This should be part of `TraceTokenRequestFilter`.

**Comment by nezihyigitbasi** [comment]
Will open a new PR as discussed offline.

**Pull request closed without merging**